File: app-miyou/src/testFixtures/java/com/miyou/app/fixture/UserCreditFixture.java
package com.miyou.app.fixture;

import com.miyou.app.domain.credit.model.UserCredit;
import com.miyou.app.domain.dialogue.model.UserId;

public final class UserCreditFixture {

	public static final long DEFAULT_BALANCE = 5000L;
	public static final long DEFAULT_VERSION = 0L;

	private UserCreditFixture() {
	}

	public static UserCredit create() {
		return UserCredit.initialize(UserIdFixture.create(), DEFAULT_BALANCE);
	}

	public static UserCredit create(UserId userId) {
		return UserCredit.initialize(userId, DEFAULT_BALANCE);
	}

	public static UserCredit create(UserId userId, long balance) {
		return new UserCredit(userId, balance, DEFAULT_VERSION);
	}

	public static UserCredit withZeroBalance(UserId userId) {
		return new UserCredit(userId, 0L, DEFAULT_VERSION);
	}

	public static UserCredit withBalance(long balance) {
		return new UserCredit(UserIdFixture.create(), balance, DEFAULT_VERSION);
	}
}

File: app-miyou/src/testFixtures/kotlin/com/miyou/app/fixture/UserCreditFixture.kt
package com.miyou.app.fixture

import com.miyou.app.domain.credit.model.UserCredit
import com.miyou.app.domain.dialogue.model.UserId

object UserCreditFixture {
    const val DEFAULT_BALANCE = 5000L
    const val DEFAULT_VERSION = 0L

    @JvmStatic
    fun create(): UserCredit = UserCredit.initialize(UserIdFixture.create(), DEFAULT_BALANCE)

    @JvmStatic
    fun create(userId: UserId): UserCredit = UserCredit.initialize(userId, DEFAULT_BALANCE)

    @JvmStatic
    fun create(
        userId: UserId,
        balance: Long,
    ): UserCredit = UserCredit(userId, balance, DEFAULT_VERSION)

    @JvmStatic
    fun withZeroBalance(userId: UserId): UserCredit = UserCredit(userId, 0L, DEFAULT_VERSION)

    @JvmStatic
    fun withBalance(balance: Long): UserCredit = UserCredit(UserIdFixture.create(), balance, DEFAULT_VERSION)
}
